vector/src/main/java/im/vector/app/features/userdirectory/UserListViewModel.kt

```diff
@@ -64,24 +77,77 @@ class UserListViewModel @AssistedInject constructor(@Assisted initialState: User
         }
     }
 
+    private val identityServerListener = object : IdentityServiceListener {
+        override fun onIdentityServerChange() {
+            withState {
+                identityServerUsersSearch.accept(it.searchTerm)
+                setState {
+                    copy(
+                            configuredIdentityServer = cleanISURL(session.identityService().getCurrentIdentityServerUrl())
+                    )
+                }
+            }
+        }
+    }
+
     init {
         observeUsers()
+        setState {
+            copy(
+                    configuredIdentityServer = cleanISURL(session.identityService().getCurrentIdentityServerUrl())
+            )
+        }
+        session.identityService().addListener(identityServerListener)
+    }
+
+    private fun cleanISURL(url: String?): String? {
+        return if (url?.startsWith("https://") == true) {
```

Comment:
might be able to use `removePrefix` to avoid the `if/substring` dance! :dancers: 

```kotlin
return url?.removePrefix("https://")
```

Reply:
👍